internal/pkg/config/v1/config.go

```diff
@@ -314,8 +388,11 @@ func (c *Config) Validate() error {
 		if _, ok := c.ContextStates[context.Name]; !ok {
 			c.ContextStates[context.Name] = new(ContextState)
 		}
-		if *c.ContextStates[context.Name] != *context.State {
+		if !reflect.DeepEqual(*c.ContextStates[context.Name], *context.State) {
 			log.CliLogger.Tracef("state of context %s in config does not match actual state of context", context.Name)
+			if c.IsTest {
```

Comment:
unit tests can't pass this test. the pre-written golden files can't know the state (with random salt and nonce in it) before hand.